darts/timeseries.py

```diff
@@ -1467,30 +1493,8 @@ def append_values(self, values: np.ndarray) -> 'TimeSeries':
 
         return self.append(TimeSeries.from_times_and_values(values=values,
                                                             times=idx,
-                                                            fill_missing_dates=False))
-
-    def update(self,
-               index: pd.DatetimeIndex,
-               values: np.ndarray = None) -> 'TimeSeries':
-        """
-        Updates the TimeSeries with the new values provided.
-        If indices are not in original TimeSeries, they will be discarded.
-        Use `numpy.nan` to ignore a specific index in a series.
-
-        Parameters
-        ----------
-        index
-            A `pandas.DateTimeIndex` containing the indices to replace.
-        values
-            An array containing the values to replace (optional).
-
-        Returns
-        -------
-        TimeSeries
-            A new TimeSeries with updated values.
-        """
-        # TODO: I don't think this is needed... probably better to just create a new TimeSeries
-        raise NotImplementedError('TimeSeries.update() is not supported anymore.')
```

Comment:
I know update is not a full new Timeseries, but sometimes it is easier to change values like the .loc[index, col] = new_value of pandas. What is the alternative to change just a small part of a timeseries? to extract the values via pd_daframe, change the values and create a new one?

Reply:
I would be interested in that as well. Just out of curiosity: would we have to implement this through `__getitem__` and `__setitem__`?

Reply:
I'm very rigid-minded about the immutability as you know :D
The "best" way today (not fully satisfying I agree) is probably to get the underlying DataArray (calling `data_array()`), edit the values as you want, and then create a new `TimeSeries` from there.
I don't want to provide functions that could mutate the series. However, we could have a function like `with_values_updated()` or something like this, which returns a _new_ TimeSeries with changes in the desired places.